timemachine/md/moves.py

```diff
@@ -12,69 +14,96 @@
 
 from timemachine import lib
 from timemachine.lib import custom_ops
-from timemachine.md.barostat.utils import get_bond_list, get_group_indices
 from timemachine.md.states import CoordsVelBox
-from timemachine.potentials import BoundPotential, HarmonicBond
+from timemachine.potentials import BoundPotential
 
 _State = TypeVar("_State")
 
 
-class MonteCarloMove(Generic[_State]):
-    n_proposed: int = 0
-    n_accepted: int = 0
+class Move(Generic[_State], ABC):
+    @abstractmethod
+    def move(self, _: _State) -> _State:
+        ...
+
+    def sample_chain_iter(self, x: _State) -> Iterator[_State]:
```

Comment:
Maybe this is a convention already, but seems to have an infinite iterator here. Would be good to have a docstring to mention this, as it would be dangerous to run this in production